### ISSUE TITLE
Dependency hell when installing Cookiecutter

### ISSUE THREAD

**Issue by diznq** [backstage/backstage#13949]
## Expected Behavior

After installing plug-in, application compiles and works

## Actual Behavior

After installing plug-in using `yarn add --cwd packages/backend @backstage/plugin-scaffolder-backend-module-cookiecutter`, application doesn't compile, because `yarn` install multiple different versions of `@backstage/integration` and `@backstage/backend-commons` at the same time, in our case before we'd only use `@backstage/integration:1.1.0`, after installing the plugin there is `1.1.0` and `1.3.1` installed at same time and one of them has `bitbucketCloud` attribute and another one doesn't, leading to compile time type error during `yarn tsc`.

## Steps to Reproduce

1. Run `yarn add --cwd packages/backend @backstage/plugin-scaffolder-backend-module-cookiecutter` 
2. Run `yarn tsc`

## Context

`packages/backend/package.json` file:

```json
{
  "name": "backend",
  "version": "0.0.0",
  "main": "dist/index.cjs.js",
  "types": "src/index.ts",
  "private": true,
  "backstage": {
    "role": "backend"
  },
  "scripts": {
    "start": "backstage-cli package start",
    "build": "backstage-cli package build",
    "lint": "backstage-cli package lint",
    "test": "backstage-cli package test",
    "clean": "backstage-cli package clean",
    "build-image": "docker build ../.. -f Dockerfile --tag backstage"
  },
  "dependencies": {
    "app": "link:../app",
    "@backstage/backend-common": "^0.13.2",
    "@backstage/backend-tasks": "^0.3.0",
    "@backstage/catalog-model": "^1.0.1",
    "@backstage/catalog-client": "^1.0.1",
    "@backstage/config": "^1.0.0",
    "@backstage/plugin-app-backend": "^0.3.31",
    "@backstage/plugin-auth-backend": "^0.13.0",
    "@backstage/plugin-catalog-backend": "^1.1.0",
    "@backstage/plugin-permission-common": "^0.6.0",
    "@backstage/plugin-permission-node": "^0.6.0",
    "@backstage/plugin-proxy-backend": "^0.2.25",
    "@backstage/plugin-scaffolder-backend": "^1.1.0",
    "@backstage/plugin-search-backend": "^0.5.0",
    "@backstage/plugin-search-backend-module-pg": "^0.3.2",
    "@backstage/plugin-search-backend-node": "^0.6.0",
    "@backstage/plugin-techdocs-backend": "^1.1.0",
    "dockerode": "^3.3.1",
    "express": "^4.17.1",
    "express-promise-router": "^4.1.0",
    "luxon": "^2.0.2",
    "pg": "^8.3.0",
    "winston": "^3.2.1"
  },
  "devDependencies": {
    "@backstage/cli": "^0.17.0",
    "@types/dockerode": "^3.3.0",
    "@types/express": "^4.17.6",
    "@types/express-serve-static-core": "^4.17.5",
    "@types/luxon": "^2.0.4"
  },
  "files": [
    "dist"
  ]
}
```

`package.json` file:

```json
{
  "name": "root",
  "version": "1.0.0",
  "private": true,
  "engines": {
    "node": "14 || 16"
  },
  "scripts": {
    "dev": "concurrently \"yarn start\" \"yarn start-backend\"",
    "start": "yarn workspace app start",
    "start-backend": "yarn workspace backend start",
    "build": "backstage-cli repo build --all",
    "build-image": "yarn workspace backend build-image",
    "tsc": "tsc",
    "tsc:full": "tsc --skipLibCheck false --incremental false",
    "clean": "backstage-cli clean && lerna run clean",
    "diff": "lerna run diff --",
    "test": "backstage-cli test",
    "test:all": "lerna run test -- --coverage",
    "lint": "backstage-cli repo lint --since origin/master",
    "lint:all": "backstage-cli repo lint",
    "prettier:check": "prettier --check .",
    "create-plugin": "backstage-cli create-plugin --scope internal",
    "remove-plugin": "backstage-cli remove-plugin"
  },
  "workspaces": {
    "packages": [
      "packages/*",
      "plugins/*"
    ]
  },
  "devDependencies": {
    "@backstage/cli": "^0.17.0",
    "@spotify/prettier-config": "^12.0.0",
    "concurrently": "^6.0.0",
    "lerna": "^4.0.0",
    "prettier": "^2.3.2",
    "typescript": "~4.5.4"
  },
  "resolutions": {
    "@types/react": "^17",
    "@types/react-dom": "^17"
  },
  "prettier": "@spotify/prettier-config",
  "lint-staged": {
    "*.{js,jsx,ts,tsx,mjs,cjs}": [
      "eslint --fix",
      "prettier --write"
    ],
    "*.{json,md}": [
      "prettier --write"
    ]
  }
}
```

Error:
```
$ yarn tsc
yarn run v1.22.19
$ tsc
packages/backend/src/plugins/scaffolder.ts:33:7 - error TS2739: Type 'ScmIntegrations' is missing the following properties from type 'ScmIntegrations': bitbucketCloud, bitbucketServer

33       integrations,
         ~~~~~~~~~~~~

  node_modules/@backstage/plugin-scaffolder-backend-module-cookiecutter/dist/index.d.ts:18:5
    18     integrations: ScmIntegrations;
           ~~~~~~~~~~~~
    The expected type comes from property 'integrations' which is declared here on type '{ reader: UrlReader; integrations: ScmIntegrations; containerRunner: ContainerRunner; }'  


Found 1 error.
```

## Your Environment

```
OS:   Windows_NT 10.0.19044 - win32/x64
node: v16.13.0
yarn: 1.22.19
cli:  0.17.0 (installed)

Dependencies:
  @backstage/app-defaults                                  1.0.1
  @backstage/backend-common                                0.13.2, 0.15.1
  @backstage/backend-plugin-api                            0.1.2
  @backstage/backend-tasks                                 0.3.0, 0.3.5
  @backstage/catalog-client                                1.0.1, 1.1.0
  @backstage/catalog-model                                 1.0.1, 1.1.1
  @backstage/cli-common                                    0.1.10, 0.1.8
  @backstage/cli                                           0.17.0
  @backstage/config-loader                                 1.1.0, 1.1.4
  @backstage/config                                        1.0.0, 1.0.2
  @backstage/core-app-api                                  1.0.1
  @backstage/core-components                               0.9.3
  @backstage/core-plugin-api                               1.0.1
  @backstage/errors                                        1.1.1, 1.0.0
  @backstage/integration-react                             1.0.1
  @backstage/integration                                   1.1.0, 1.3.1
  @backstage/plugin-api-docs                               0.8.4
  @backstage/plugin-app-backend                            0.3.31
  @backstage/plugin-auth-backend                           0.13.0
  @backstage/plugin-auth-node                              0.2.0, 0.2.5
  @backstage/plugin-catalog-backend                        1.1.1, 1.4.0
  @backstage/plugin-catalog-common                         1.0.1, 1.0.6
  @backstage/plugin-catalog-graph                          0.2.16
  @backstage/plugin-catalog-import                         0.8.7
  @backstage/plugin-catalog-node                           1.1.0
  @backstage/plugin-catalog-react                          1.0.1
  @backstage/plugin-catalog                                1.1.0
  @backstage/plugin-github-actions                         0.5.4
  @backstage/plugin-org                                    0.5.4
  @backstage/plugin-permission-common                      0.6.0, 0.6.4
  @backstage/plugin-permission-node                        0.6.0, 0.6.5
  @backstage/plugin-permission-react                       0.4.0
  @backstage/plugin-proxy-backend                          0.2.25
  @backstage/plugin-scaffolder-backend-module-cookiecutter 0.2.11
  @backstage/plugin-scaffolder-backend                     1.1.0, 1.6.0
  @backstage/plugin-scaffolder-common                      1.0.1, 1.2.0
  @backstage/plugin-scaffolder                             1.1.0
  @backstage/plugin-search-backend-module-pg               0.3.2
  @backstage/plugin-search-backend-node                    0.6.0
  @backstage/plugin-search-backend                         0.5.1
  @backstage/plugin-search-common                          0.3.3, 1.0.1
  @backstage/plugin-search-react                           0.1.0
  @backstage/plugin-search                                 0.8.0
  @backstage/plugin-tech-radar                             0.5.11
  @backstage/plugin-techdocs-backend                       1.1.0
  @backstage/plugin-techdocs-node                          1.1.0
  @backstage/plugin-techdocs-react                         0.1.0
  @backstage/plugin-techdocs                               1.1.0
  @backstage/plugin-user-settings                          0.4.3
  @backstage/release-manifests                             0.0.2
  @backstage/search-common                                 0.3.3
  @backstage/test-utils                                    1.0.1
  @backstage/theme                                         0.2.15
  @backstage/types                                         1.0.0
  @backstage/version-bridge                                1.0.1
Done in 1.80s.
```

**Comment by nadworny** [comment]
hi @OrkoHunter , we've been using cookiecutter for years before we started using backstage templates and we don't want to migrate all of our templates especially since we're using a lot of cookiecutter-specific methods which aren't covered by nunjucks.

**Comment by freben** [comment]
Indeed, the error seems to be because you installed a version of that plugin that was newer than the rest of your Backstage installation, so you ended up with duplicates of packages. You may be able to run `yarn backstage-cli versions:check` to see where your duplicates are, or go directly to `yarn why @backstage/integration` to see what pulls in what.

One option is to upgrade Backstage before pulling in the plugin, or maybe we could help you find an older version of the plugin that was built for your Backstage.

**Comment by Rugvip** [comment]
Want to highlight also that you can look up previous package versions over at https://versions.backstage.io/v1/releases/1.1.0/manifest.json. Check the release version in your `backstage.json` and then replace `1.1.0` in the URL with your own version, and you'll get a list of all package versions from that release.

**Comment by nadworny** [comment]
I think this can be closed?

**Comment by Rugvip** [comment]
Indeed, thanks!

**Comment by regicsolutions** [comment]
> @diznq Aah that is unfortunate. That is one of the reasons why we made `cookiecutter` an optional templating engine and implemented a native nunjucks based node templating.
> 
> 
> 
> Is there any particular reason you're planning to use cookiecutter over the default templating engine? Just curious. :)

@OrkoHunter i have a use case where I need to run golang cli commands which requires golang as a dependency, Was hoping to make use of the out of the box nunjucks templating using the fetch:template action but need to be able to run that action inside a golang container, looks like the cookiecutter module makes use of containerRunner, do you know if this is possible using fetch:template? Any docs you can point me to would be greatly appreciated!
